x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/tree/EsqlNodeSubclassTests.java

```diff
@@ -58,13 +139,276 @@ public class EsqlNodeSubclassTests<T extends B, B extends Node<B>> extends NodeS
+                return EnumSet.of(enm);
+            }
+            if (toBuildClass == OutputExec.class && pt.getRawType() == Consumer.class) {
+                // pageConsumer just needs a BiConsumer. But the consumer has to have reasonable
+                // `equals` for randomValueOtherThan, so we just ensure that a new instance is
+                // created each time which uses Object::equals identity.
+                return new Consumer<Page>() {
+                    @Override
+                    public void accept(Page page) {
+                        // do nothing
+                    }
+                };
+            }
+
+            throw new IllegalArgumentException("Unsupported parameterized type [" + pt + "], for " + toBuildClass.getSimpleName());
+        }
+        if (argType instanceof WildcardType wt) {
+            if (wt.getLowerBounds().length > 0 || wt.getUpperBounds().length > 1) {
+                throw new IllegalArgumentException("Unsupported wildcard type [" + wt + "]");
+            }
+            return makeArg(toBuildClass, wt.getUpperBounds()[0]);
+        }
+        Class<?> argClass = (Class<?>) argType;
+
+        /*
+         * Sometimes all of the required type information isn't in the ctor
+         * so we have to hard code it here.
+         */
+        if (toBuildClass == FieldAttribute.class) {
+            // `parent` is nullable.
+            if (argClass == FieldAttribute.class && randomBoolean()) {
+                return null;
+            }
+        } else if (toBuildClass == NodeTests.ChildrenAreAProperty.class) {
+            /*
+             * While any subclass of DummyFunction will do here we want to prevent
+             * stack overflow so we use the one without children.
+             */
+            if (argClass == NodeTests.Dummy.class) {
+                return makeNode(NodeTests.NoChildren.class);
+            }
+        } else if (FullTextPredicate.class.isAssignableFrom(toBuildClass)) {
+            /*
+             * FullTextPredicate analyzes its string arguments on
+             * construction so they have to be valid.
+             */
+            if (argClass == String.class) {
+                int size = between(0, 5);
+                StringBuilder b = new StringBuilder();
+                for (int i = 0; i < size; i++) {
+                    if (i != 0) {
+                        b.append(';');
+                    }
+                    b.append(randomAlphaOfLength(5)).append('=').append(randomAlphaOfLength(5));
+                }
+                return b.toString();
+            }
+        } else if (toBuildClass == Like.class) {
+
+            if (argClass == LikePattern.class) {
+                return new LikePattern(randomAlphaOfLength(16), randomFrom('\\', '|', '/', '`'));
+            }
+
+        } else if (argClass == Dissect.Parser.class) {
             // Dissect.Parser is a record / final, cannot be mocked
```

Comment:
++ Nice not to have to jump around between `makeArg` and `pluggableMakeArg` when debugging anymore.